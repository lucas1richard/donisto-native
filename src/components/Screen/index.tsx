import * as React from 'react';
import { SafeAreaView } from 'react-native';
import { SafeAreaViewProps } from 'react-navigation';

export interface IScreenProps {
  style?: SafeAreaViewProps
}

class Screen extends React.Component<IScreenProps> {
  render() {
    const { children, style = {} } = this.props;
    return (
      <SafeAreaView style={{flex: 1, ...style }}>
        {children}
      </SafeAreaView>
    );
  }
}

export default Screen;
