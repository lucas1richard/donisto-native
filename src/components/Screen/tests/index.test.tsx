import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Screen from '..';

describe('<Screen />', () => {
  const renderComponent = () => toJson(shallow(<Screen />));
  it('renders', () => {
    expect(renderComponent()).toMatchSnapshot();
  });
});