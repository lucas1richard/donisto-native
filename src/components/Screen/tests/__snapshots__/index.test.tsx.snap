// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> renders 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
